kdc/kx509.c

```diff
@@ -894,10 +892,12 @@ _kdc_do_kx509(kx509_req_context r)
     /* Authenticate the request (consume the AP-REQ) */
     ret = krb5_kt_resolve(r->context, "HDBGET:", &id);
     if (ret) {
+	char *msg = krb5_get_error_message(r->context, ret);
```

Comment:
`krb5_get_error_message()` returns a `const char *`.  It does need freeing, yes, but with `krb5_free_error_message()`, not `free()`.